pkg/services/ngalert/state/historian/loki_http.go

```diff
@@ -199,8 +199,12 @@ func (c *httpLokiClient) query(ctx context.Context, selectors []Selector, start,
 
 	values := url.Values{}
 	values.Set("query", selectorString(selectors))
-	values.Set("start", fmt.Sprintf("%d", start))
-	values.Set("end", fmt.Sprintf("%d", end))
+	if start != 0 {
+		values.Set("start", fmt.Sprintf("%d", start))
+	}
+	if end != 0 {
+		values.Set("end", fmt.Sprintf("%d", end))
+	}
```

Comment:
I'd document the behaviour of range query defaults here - When no start or end timestamps are sent, Loki will default to `1h` for the start timestamp and `now` for the end time.

One more thing that I don't like is that start and end are kind-of like options parameters but you must specify something at the function call - can we at least document the function, it so that users of the method understand that behaviour?

Reply:
Perhaps it might make sense to use a `QueryOptions` struct instead?

```go
type QueryOptions struct {
    selectors []Selector
    start     int64 // an optional start time
    end       int64 // an optional end time
}
```

Reply:
Looking at this with fresh eyes, I took a new approach that I think addresses all these.

Now, we move this up a layer - the client still requires timestamps, but we calculate our own defaults instead.

1. The layer above already uses a query options struct like in @grobinson-grafana's suggestion.
2. No need for docs to explain weird edge case behavior at the client level. Code is more clear.
3. 1 hour is honestly not a good default for state history. It's too short. In the new approach, we can add our own default.
4. Using Grafana's `time.Now()` rather than Loki's is generally more correct in the event that system clocks do not align.
5. Better single responsibility principle - let's keep `loki_http.go` focused on the actual HTTP communication, the default range is app logic that's agnostic of transport.

Reply:
I'm with George - `QueryOptions` is what I would expect. Aligns with open/close and SRP.

Reply:
Ah, nvm - I see what you mean with "moving up a layer". LGTM.